Boto3 and boto fixup

willthames opened the pull request:
##### SUMMARY
Change all modules that import both boto and boto3

Update to use latest connection handling (so no need to check for region any more, boto3_conn does this)

Some pep8 tidy up

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
Various AWS modules

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 5ff36c3423) last updated 2017/11/10 12:20:32 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, Sep  5 2017, 08:53:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```


##### ADDITIONAL INFORMATION

These modules are marked as preview so no tests are required. I'll add some information about what testing I've performed.

willthames committed:
Remove boto usage from boto3 modules

willthames commented:
I've tested ecs_cluster, ecs_service and ecs_task_definition following these changes with little adverse effect, but welcome any further testing

gundalow commented:
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

willthames committed:
Remove region checking

boto3_conn now takes care of region checking and handles NoRegionError
exceptions with a standard message

boto3_conn also takes care of other connection exceptions too.

s-hertel committed:
Document boto3 as a requirement for ec2_eni_facts
